src/commands/cmd_hash.cc

```diff
@@ -376,6 +376,54 @@ class CommandHScan : public CommandSubkeyScanBase {
     return Status::OK();
   }
 };
+class CommandHRandField : public Commander {
+ public:
+  Status Parse(const std::vector<std::string> &args) override {
+    if (args.size() >= 3) {
+      noParameters_ = false;
+      auto parse_result = ParseInt<int64_t>(args[2], 10);
+      if (!parse_result) {
+        return {Status::RedisParseErr, errValueNotInteger};
+      }
+      l_ = *parse_result;
+
+      if (args.size() > 4 || (args.size() == 4 && strcasecmp(args[3].c_str(), "withvalues"))) {
+        return {Status::RedisParseErr, errInvalidSyntax};
+      } else if (args.size() == 4) {
+        withvalues_ = true;
+      }
+    }
+    return Commander::Parse(args);
+  }
+  Status Execute(Server *svr, Connection *conn, std::string *output) override {
+    redis::Hash hash_db(svr->storage, conn->GetNamespace());
+    std::vector<FieldValue> field_values;
+
+    auto s = hash_db.RandField(args_[1], &field_values, l_, noParameters_,
+                               withvalues_ ? HashFetchType::kAll : HashFetchType::kOnlyKey);
+    if (!s.ok() && !s.IsNotFound()) {
+      return {Status::RedisExecErr, s.ToString()};
+    }
+
+    std::vector<std::string> result_entries;
+    result_entries.reserve(field_values.size());
+    for (const auto &p : field_values) {
+      result_entries.emplace_back(p.field);
+      if (withvalues_) result_entries.emplace_back(p.value);
+    }
+
+    if (noParameters_)
+      *output = s.IsNotFound() ? redis::NilString() : redis::BulkString(result_entries[0]);
+    else
+      *output = redis::MultiBulkString(result_entries, false);
+    return Status::OK();
+  }
+
+ private:
+  bool withvalues_ = false;
+  int64_t l_ = 0;
```

Comment:
`l_` is a confusing name, I guess `count_` could be much better

Reply:
Hmm I guess Redis Command level uses name `COUNT`. Maybe using `command_count` or other is ok?